File: PDFExporterDemo/Views/FooterView.m
//
//  FooterView.m
//  PDFExporterDemo
//
//  Copyright © 2015 3Pillar Global. All rights reserved.
//

#import "FooterView.h"

static NSString * const kPageLabelFormat = @"%lu / %lu";

@interface FooterView ()

@property (weak, nonatomic) IBOutlet UILabel *pageLabel;

@end

@implementation FooterView

- (instancetype)initFromXib {
    self = [self init];
    
    if (self) {
        NSArray<UIView *> *xibViews = [[NSBundle mainBundle] loadNibNamed:NSStringFromClass([self class])
                                                                    owner:self
                                                                  options:nil];
        UIView *view = [xibViews firstObject];
        self.bounds = view.bounds;
        [self addSubview:view];
    }
    
    return self;
}

- (void)updatePageNumber:(NSUInteger)pageNumber totalPages:(NSUInteger)totalPages {
    self.pageLabel.text = [NSString stringWithFormat:kPageLabelFormat, (unsigned long)pageNumber, (unsigned long)totalPages];
}

@end

File: Example/PDFExporterDemo/PDFExporterDemo/View/FooterView.swift
//
//  FooterView.swift
//  PDFExporterDemo
//
//  Created by Ciprian Antal on 07/12/2017.
//  Copyright © 2017 3PG. All rights reserved.
//

import UIKit
import PDFExporter

class FooterView: UIView, PDFHeaderFooterPaging {
    @IBOutlet weak var pageLabel: UILabel!

    class func instanceFromNib() -> FooterView? {
        guard let footerView =  UINib(nibName: "FooterView",
                                      bundle: nil).instantiate(withOwner: nil, options: nil).first as? FooterView
            else { return nil }
        return footerView
    }

    func updatePageNumber(_ pageNumber: UInt, totalPages: UInt) {
         pageLabel.text = "\(pageNumber)/\(totalPages)"
    }
}
